File: src/netlivecowork/api/schemas/skills.py
"""Skill management response models (match frontend-desktop/src/api/skills.ts)."""

from __future__ import annotations

from pydantic import BaseModel


class LocalSkillResponse(BaseModel):
    skill_id: str
    name: str
    description: str
    version: str
    triggers: list[str]
    origin: str = "local"            # "local"（自建，永久存）| "cloud"（市场引用，用时下载）
    source: str | None = None        # 仅 cloud：来源市场 "cowork" | "mythos"（前端徽章用）
    # 归属：这条 skill 归哪些 cowork 用。``["*"]`` = 通用（谁都能用）。
    #
    # ⚠ **通用的也要显示**（需求 H7）：用户手里多数是通用 skill，
    # 通用不占位的话整个"归属"概念在界面上根本不存在，人就无从判断
    # 自己刚才的选择有没有生效。
    coworks: list[str] = ["*"]


class SkillMarketTab(BaseModel):
    """技能市场的一个页签。``cowork=None`` 是通用市场（不属于任何 cowork，恒存在）。"""
    cowork: str | None = None
    display_name: str


class RemoteCatalogItem(BaseModel):
    source: str               # "cowork" | "mythos" —— 程序用于下载路由，UI 不展示
    id: str
    #: 这条目录项在**当前页签作用域**下的确定性引用 ID（不透明字符串）。
    #: 前端据它与已引用列表精确配对：同 source/id 在通用与专属市场是两条不同的引用。
    reference_id: str
    name: str
    description: str | None
    updater: str | None = None            # 作者。各家字段名不同（netcowork 回 creatorName），由 adapter归一
    #: 下载/引用次数。**必须在这里声明**——response_model 没有的字段 FastAPI 会直接丢掉，
    #: 表现是后端算出来了、接口里却没有，而两边都不报错。
    download_count: int | None = None
    create_time: str | None
    is_pulled: bool


class PullSkillResponse(BaseModel):
    skill_id: str
    name: str

File: frontend-desktop-v2/src/api/skills.ts
import { http } from './client'

export interface LocalSkill {
  skill_id: string
  name: string
  description: string
  version: string
  triggers: string[]
}

// source 标明这条 skill 来自哪个市场（cowork / mythos）。用户不感知，程序用它
// 决定下载时调哪个数据源的接口。
export type SkillSource = 'cowork' | 'mythos'

export interface RemoteCatalogItem {
  source: SkillSource
  id: string
  name: string
  description: string | null
  updater: string | null
  create_time: string | null
  is_pulled: boolean
}

export interface PullSkillResponse {
  skill_id: string
  name: string
}

export const skillsApi = {
  list:         () => http.get<LocalSkill[]>('/skills'),
  delete:       (skillId: string) => http.delete<void>(`/skills/${skillId}`),
  importLocal:  (file: File) => http.upload<LocalSkill>('/skills/import', file),

  // username 给 mythos 用（其请求头要带当前登录用户名）；cowork 忽略它。
  catalog:      (username: string) =>
    http.get<RemoteCatalogItem[]>(`/skills/pull-server/catalog?username=${encodeURIComponent(username)}`),
  // 安装时把这条 skill 的 source 原样回传，后端据此派发到对应数据源的下载接口。
  pull:         (item: { id: string; name: string; source: SkillSource }, username: string) =>
    http.post<PullSkillResponse>(`/skills/pull-server/catalog/${item.id}/pull`, {
      name: item.name, source: item.source, username,
    }),
  importRemote: (file: File) => http.upload<PullSkillResponse>('/skills/pull-server/import', file),
}
